### PR TITLE
Internal: Packages - Build without link

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: .grunt-config/webpack.packages.js [hunk 1]
```diff
@@ -87,3 +89,53 @@ module.exports = {
...
+			path: path.resolve( __dirname, `../node_modules/${ packageName }`, mainFile ),
+		} ) );
+}
+
+function getPackagesBasedOnLocalRepo() {
+	const repoPath = process.env.ELEMENTOR_PACKAGES_PATH;
+	const relevantDirs = [ 'packages/core', 'packages/libs' ]
+
+	if ( ! repoPath ) {
+		throw new Error('ELEMENTOR_PACKAGES_REPO_PATH is not defined' );
```
Comment (StyleShit) [inline .grunt-config/webpack.packages.js:120]:
```suggestion
		throw new Error('ELEMENTOR_PACKAGES_PATH is not defined' );
```

File: .grunt-config/webpack.packages.js [hunk 2]
```diff
@@ -87,3 +89,53 @@ module.exports = {
...
+			name: packageName.replace( '@elementor/', '' ),
+			path: path.resolve( __dirname, `../node_modules/${ packageName }`, mainFile ),
+		} ) );
+}
+
+function getPackagesBasedOnLocalRepo() {
+	const repoPath = process.env.ELEMENTOR_PACKAGES_PATH;
+	const relevantDirs = [ 'packages/core', 'packages/libs' ]
+
+	if ( ! repoPath ) {
```
Comment (StyleShit) [inline .grunt-config/webpack.packages.js:119]: also check if it exists?

File: .grunt-config/webpack.packages.js [hunk 3]
```diff
@@ -87,3 +89,53 @@ module.exports = {
...
+		.filter( ( { mainFile } ) => !! mainFile )
+		.map( ( { mainFile, packageName } ) => ( {
+			name: packageName.replace( '@elementor/', '' ),
+			path: path.resolve( __dirname, `../node_modules/${ packageName }`, mainFile ),
+		} ) );
+}
+
+function getPackagesBasedOnLocalRepo() {
+	const repoPath = process.env.ELEMENTOR_PACKAGES_PATH;
+	const relevantDirs = [ 'packages/core', 'packages/libs' ]
```
Comment (StyleShit) [inline .grunt-config/webpack.packages.js:117]: hmm, do we need tools here?
is there any case we would want to use only specific package from the local and others from npm?

File: package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
 		"test:jest": "jest --config='./tests/jest/jest.config.js'",
 		"scripts": "grunt scripts",
 		"scripts:watch": "grunt watch_scripts",
+		"scripts:packages:local": "export ELEMENTOR_PACKAGES_USE_LOCAL=1 && grunt scripts:packages:local",
```
Comment (StyleShit) [inline package.json:36]:
```suggestion
		"scripts:packages:local": "cross-env ELEMENTOR_PACKAGES_USE_LOCAL=1 && grunt scripts:packages:local",
```

Maybe also add "npm run scripts:packages"

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: .grunt-config/webpack.packages.js (hunk 1 above)
Comment (Nevoss) [reply .grunt-config/webpack.packages.js:120]: Done

File: .grunt-config/webpack.packages.js (hunk 2 above)
Comment (Nevoss) [reply .grunt-config/webpack.packages.js:119]: Done

File: .grunt-config/webpack.packages.js (hunk 3 above)
Comment (Nevoss) [reply .grunt-config/webpack.packages.js:117]: no

File: package.json (hunk 4 above)
Comment (Nevoss) [reply package.json:36]: Done

File: .grunt-config/webpack.packages.js [hunk 5]
```diff
@@ -87,3 +89,53 @@ module.exports = {
...
+
+	if ( ! repoPath && fs.existsSync( repoPath ) ) {
```
Comment (StyleShit) [inline .grunt-config/webpack.packages.js:119]:
```suggestion
	if ( ! repoPath || ! fs.existsSync( repoPath ) ) {
```

I would also consider putting it in a different condition and different error message, I think it's much better DX

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: .grunt-config/webpack.packages.js (hunk 5 above)
Comment (Nevoss) [reply .grunt-config/webpack.packages.js]: Done

Comment (Nevoss) [reply .grunt-config/webpack.packages.js]: My bad

**Review by StyleShit — DISMISSED** [review]

File: .grunt-config/webpack.packages.js [hunk 6]
```diff
@@ -87,3 +89,57 @@ module.exports = {
 	dev: devConfig,
 	prod: prodConfig,
 };
+
+function getPackagesBasedOnProjectDeps() {
```
Comment (rotemee) [inline .grunt-config/webpack.packages.js:93]: "getPackagesBasedOnProjectDeps" is a bit confusing, maybe just describing the function output?: "getElementorPackagesMainFiles" or something like that?

File: .grunt-config/webpack.packages.js [hunk 7]
```diff
@@ -87,3 +89,57 @@ module.exports = {
...
+			}
+		} )
+		.filter( ( { mainFile } ) => !! mainFile )
+		.map( ( { mainFile, packageName } ) => ( {
+			name: packageName.replace( '@elementor/', '' ),
+			path: path.resolve( __dirname, `../node_modules/${ packageName }`, mainFile ),
+		} ) );
+}
+
+function getPackagesBasedOnLocalRepo() {
```
Comment (rotemee) [inline .grunt-config/webpack.packages.js:115]: Same here, maybe "getLocalReposPackagesMainFiles"?

**Review by rotemee — CHANGES_REQUESTED** [review]

File: .grunt-config/webpack.packages.js [hunk 8]
```diff
@@ -87,3 +89,57 @@ module.exports = {
...
+	const packages = relevantDirs.flatMap( ( dir ) => {
+		const dirPath = path.resolve( repoPath, dir );
+		const packages = fs.readdirSync( path.resolve( repoPath, dir ) )
+
+		return packages.map(( name ) => {
+			return {
+				name,
+				path: path.resolve( dirPath, `${name}/src/index.ts` ),
+			}
+		})
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply .grunt-config/webpack.packages.js]: Done

File: .grunt-config/webpack.packages.js (hunk 7 above)
Comment (Nevoss) [reply .grunt-config/webpack.packages.js]: Changed

File: .grunt-config/webpack.packages.js (hunk 6 above)
Comment (Nevoss) [reply .grunt-config/webpack.packages.js]: Changed

**Review by rotemee — APPROVED** [review]

**Pull request merged**